src/Control/Session.php

```diff
@@ -300,6 +300,7 @@ public function start(HTTPRequest $request)
 
         // If the session cookie is already set, then the session can be read even if headers_sent() = true
         // This helps with edge-case such as debugging.
+        $data = [];
         if (!session_id() && (!headers_sent() || !empty($_COOKIE[ini_get('session.name')]))) {
             if (!headers_sent()) {
```

Comment:
It looks like `session_start` should be in this block as if the headers have already been sent, we can't start the session and will get a PHP warning alike
```
Warning: session_start(): Cannot start session when headers already sent in ...
```

Reply:
Ok, I think i've resolved this issue too, although I noticed the `$_COOKIE[ini_get('session.name')]` wasn't checking for the secure cookie, so i've amended that by calling the helper function `requestContainsSessionId in the class instead. -  I think that's what you intended @sminnee  ?